### ISSUE TITLE
The config option `module_enable` doesn't work as expected

### ISSUE THREAD

**Issue by praiskup** [rpm-software-management/mock#232]
I prepared `/etc/mock/fedora_modular-rawhide-x86_64.cfg` config file (something similar which is requested in #231 ), but mock seems to not do what I'd expected.

Actual results:
```shell
$ mock -r fedora_modular-rawhide-x86_64 --config-opts module_enable=postgresql:9.6 --config-opts module_enable= --install postgresql-server
...
fedora                                                                                                                                                                                                       780 kB/s |  61 MB     01:20    
Fedora - Modular Rawhide - Developmental packages for the next Fedora release                                                                                                                                511 kB/s | 2.2 MB     00:04    
Last metadata expiration check: 0:00:01 ago on Fri 18 Jan 2019 11:29:45 AM CET.
Modular dependency problems:

 Problem 1: conflicting requests
  - nothing provides module(platform:f30) needed by module stratis:1:20181215204600:a5b0195c-0.x86_64
 Problem 2: conflicting requests
  - nothing provides module(platform:f30) needed by module ripgrep:latest:20181215204810:a5b0195c-0.x86_64
 Problem 3: conflicting requests
  - nothing provides module(platform:f30) needed by module ninja:latest:20181208134839:a5b0195c-0.x86_64
...
```

However, when I do:
```shell
$ mock -r fedora_modular-rawhide-x86_64 --init
...
$ mock -r fedora_modular-rawhide-x86_64 --config-opts module_enable=postgresql:9.6 --config-opts module_enable= --install postgresql-server
...
Installing:
 postgresql-server                                             x86_64                                            11.1-2.fc30                                                         fedora                                            5.1 M
...
```

Which is neither correct.  It isn't installing package from module - note the versions **9.6** vs **11**.

**Comment by praiskup** [comment]
Well in my use-case I don't plan to install server/client profile (server == the old "default").  I plan to install `postgresql-server-devel` and libs maybe.  But I'll try this.

**Comment by praiskup** [comment]
The `module_install` neither helps.

**Comment by owtaylor** [comment]
The problem is that mock runs 'dnf module enable' before fedora-release is installed by `chroot_setup_cmd`, but dnf wants to use `/etc/os-release` to get the platform name:stream. Setting `module_platform_id` in the dnf config would work, except for https://github.com/rpm-software-management/dnf/pull/1305.

I'm not sure what the right resolution is - it's a tricky chicken-and-the-egg situation -  `chroot_setup_cmd` is needed to get the right platform name:stream, but maybe you want to install packages from modules as part of `chroot_setup_cmd`. The simplest thing is to  reverse the order. the order and disallow installing packages from modules in `chroot_setup_cmd`. The alternative would be to require  the platform name:stream in the mock config and either wait for the fix for the dnf bug, or hack a fake `/etc/os-release`.

**Comment by xsuchy** [comment]
What is `module_platform_id`? @owtaylor can you point me to some documentation?

**Comment by xsuchy** [comment]
This should be resolved by c402408 and 3b3dbec
I tried mock-core-configs from devel branch and it works for me. I will appreciate any testing from you.

**Comment by praiskup** [comment]
Works, thank you!

Note that Fedora Rawhide has some temporary issues (branching time?), but this works as expected:

`mock -r fedora-30-x86_64 --config-opts module_enable=postgresql:9.6 --config-opts module_enable= --install postgresql-server`
